t/cli/test_admin.sh

```diff
@@ -189,6 +189,62 @@ fi
 
 echo "pass: missing admin key and only allow 127.0.0.0/24 to access admin api"
 
+# allow any IP to access admin api with empty admin_key, when admin_key_required=true
+
+git checkout conf/config.yaml
+
+echo '
+deployment:
+  admin:
+    admin_key_required: true
+    admin_key: ~
+    allow_admin:
+      - 0.0.0.0/0
+' > conf/config.yaml
+
+make init > output.log 2>&1 | true
+
+if ! grep -E "ERROR: missing valid Admin API token." output.log > /dev/null; then
+    echo "failed: should show 'ERROR: missing valid Admin API token.'"
+    exit 1
+fi
+
+echo '
+deployment:
+  admin:
+    admin_key_required: false
+    admin_key: ~
+    allow_admin:
+      - 0.0.0.0/0
+' > conf/config.yaml
+
+make init > output.log 2>&1 | true
+
+if grep -E "ERROR: missing valid Admin API token." output.log > /dev/null; then
+    echo "failed: should not show 'ERROR: missing valid Admin API token.'"
+    exit 1
+fi
+
+if ! grep -E "Warning! AdminKey is bypassed" output.log > /dev/null; then
+    echo "failed: should show 'Warning! AdminKey is bypassed'"
+    exit 1
+fi
+
+echo '
+deployment:
+  admin:
+    admin_key_required: invalid-value
+' > conf/config.yaml
+
+make init > output.log 2>&1 | true
+
+if grep -E "path[deployment->admin->admin_key_required] expect: boolean, but got: string" output.log > /dev/null; then
+    echo "failed: should show 'expect: boolean, but got: string'"
```

Comment:
echo "check admin_key_required value failed: should show 'expect: boolean, but got: string'"

Reply:
This is the schema check error msg automatically created... It seems that it cannot be modified by custom.

Reply:
The warning message prefix is needed: `check admin_key_required value failed`, or you will not what happened only with `failed: should show 'expect: boolean, but got: string`

Reply:
I mean that this msg is automatically created and outputed by the schema check step.

Total output:
![image](https://user-images.githubusercontent.com/24536972/228785761-ed720ede-7cc2-4761-a1ab-56c411108b73.png)

It seems that we cannot easily custom info...

I think the grep text now has enough to help to know "what happened".

![image](https://user-images.githubusercontent.com/24536972/228786028-072db7b7-3e3b-49be-b5f4-3defa83d6d47.png)

Reply:
OK, this is for the output file, i mean what you `echo` to the screen

Reply:
Got it.

Reply:
Done.